starknet-replay/src/storage/mod.rs

```diff
@@ -65,7 +65,7 @@ pub trait Storage {
     fn get_transactions_and_receipts_for_block(
         &self,
         block_number: BlockNumber,
-    ) -> Result<(Vec<Transaction>, Vec<TransactionReceipt>), DatabaseError>;
+    ) -> Result<(BlockHeader, Vec<Transaction>, Vec<TransactionReceipt>), DatabaseError>;
```

Comment:
Okay this result type needs to be a named type. It's too big at this point to be ergonomic and understandable.